librouteros/connections.py

```diff
@@ -22,7 +22,13 @@ def read(self, length):
         """
         data = bytearray()
         while len(data) != length:
-            data += self.sock.recv((length - len(data)))
+            tmp = None
+            try:
+                tmp = self.sock.recv((length - len(data)))
+            except:
+                raise ConnectionClosed('Socket recv failed.')
```

Comment:
It doesn't always mean that connection was closed. I didn't wrap socket errors because you can catch them in your code.

Reply:
That was what I tried at first, but it did not work for me.
I can try again after all changes I did in my code if you prefer, but problem is that I could never reproduce this issue on my system. It was only reported by other people.